### DISCUSSION TITLE
What is turbod?

### DISCUSSION THREAD

**Discussion by paulius-valiunas** [vercel/turborepo#2117] [Help] [answered]
I'm getting a warning "failed to contact turbod". I can't find any references to turbod in the documentation, so what does it even mean? What's the purpose of turbod and where can I find and enable it?

**Comment by gkn** [comment]
Lots of info in this issue: [WARNING failed to contact turbod.](https://github.com/vercel/turborepo/issues/2034)

**Reply by paulius-valiunas** [reply]
Thanks, I saw the issue, but it doesn't really provide any information on what the daemon is for. After reading this issue I became even more confused, which is what led me to opening this discussion in the first place. I see people in that issue are trying to debug a connection between turbo and turbod, but none of that discussions answers the most important questions:
- What is the daemon for?
- Do I need it?
- How is it supposed to be started? Is there a startup script that is installed by turbo? Do I have to run it manually?

It feels like I'm reading a deeply technical discussion without having the context necessary to understand it.

I think all of these answers should be in your documentation, because honestly, I find all of this very, _very_ confusing.

**Reply by gkn** [reply]
I asked the same question a little while back, and indeed there is an answer in that thread :)
[https://github.com/vercel/turborepo/issues/2034#issuecomment-1263789573](https://github.com/vercel/turborepo/issues/2034#issuecomment-1263789573)

**Reply by paulius-valiunas** [reply]
ah, now there is :) there wasn't when I opened this discussion, I guess I should have checked for updates more carefully. It still doesn't explain how and when I should run the daemon, though.

**Reply by chris-olszewski** [reply]
TL;DR you shouldn't need to manually manage the daemon. You can manually start/stop/restart the daemon using the commands listed in [this comment](https://github.com/vercel/turborepo/issues/2034#issuecomment-1256614777).

**Reply by AddictArts** [reply]
> TL;DR you shouldn't need to manually manage the daemon. You can manually start/stop/restart the daemon using the commands listed in [this comment](https://github.com/vercel/turborepo/issues/2034#issuecomment-1256614777).

Unless the daemon doesn’t work at all, then you’re forced to `—no-daemon`

**Comment by gkn** [comment]
Please note that I am just another user of turborepo, but AFAIK it should just work automatically - except for the bugs that is :)

**Reply by AddictArts** [reply]
At least for windows users version 1.5.6 does not resolve the issue.

**Comment by mehulkar** [comment, marked as answer]
`turbod` is the turbo daemon. It is started to help optimize work that can happen in the background. The code is https://github.com/vercel/turborepo/tree/main/cli/internal/daemon. There isn't a lot of documentation as it's an implementation detail, but we document how to turn it off here https://turborepo.org/docs/reference/command-line-reference#--no-daemon
